Added two example template scripts

jbungenerflex opened the pull request:
Added two examples scripts:

1. template to upload a mesh and run 1 or many cases in a sweep.
2. template to download results for a given case.

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,129 @@
+                fl.Freestream(
+                    surfaces=mesh_object["BOUNDARY4"], name="Boundary4"
+                ),  # for far field boundaries
+                # define what sort of physical model of a fluid we will use
+                fl.Fluid(
+                    navier_stokes_solver=fl.NavierStokesSolver(),
+                    turbulence_model_solver=fl.SpalartAllmaras(),
+                ),
```
Those are default settings and there is no need to explicitly call them.
```suggestion

```

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,129 @@
+            outputs=[
+                fl.VolumeOutput(output_format="tecplot", output_fields=vol_fields),
+                # This mesh_object['*'] will select all the boundaries in the mesh and export the surface results.
+                # Regular expressions can be used to fileter for certain boundaries
```
```suggestion
                # Regular expressions can be used to filter for certain boundaries
```

wiktor-flex reviewed (commented):
All comments in the scripts should be made in uniform style, for example lowercase but starting with a capital letter like regular sentences.

If you have any questions feel free to reach out to me

jbungenerflex commented:
@wiktor-flex , I have pushed my changes. please review. Thanks

wiktor-flex commented on examples/retrieve_results/template_script_download_results.py:
```diff
@@ -0,0 +1,56 @@
+
+case_id = "ENTER CASE ID HERE"
+
+# Used only if we are interested in downloading specific data sets
+download_surfaces = True
+download_volumes = True
```
Why are we creating variables that remain unassigned until lines 35-40 get uncommented? Wouldn't it be better to assign True values directly there?

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,146 @@
+"""
+Sample Flow 360 API scripts
+
+Requires that you have a mesh that you are ready to upload and run cases on.
+
+"""
```
```suggestion
"""
Sample Flow 360 API scripts.
Requires a mesh that you are ready to upload and run cases on.
"""

```

wiktor-flex commented on examples/retrieve_results/template_script_download_results.py:
```diff
@@ -0,0 +1,56 @@
+"""
+Download volume, surface,acoustic etc... files to the directory it is run from.
+"""
```
```suggestion
"""
Download volume, surface, acoustic and other types of output files to the directory it is run from.
"""

```

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,146 @@
+
+######################################################################################################################
+def upload_mesh(file_path, project_name):
+    """
+    given a file path it will uplaod a mesh
+    Returns
+    -------
+    project id associated with that mesh
+    """
```
```suggestion
    """
    Given a file path and name of the project, this function creates a project and uploads a mesh.
    """

```

wiktor-flex reviewed (commented):
print(...) statements upon importing
`from flow360.log import log`
can be replaced with the following
`log.info("text to display")`
But this is not necessary and just an alternative that may provide better visibility and showcase additional functionality.

wiktor-flex commented:
Also, make sure that all checks are passing for you branch, here black is failing and can be fixed by running `black .`.
Here is more information about check in.
https://github.com/flexcompute/Flow360?tab=readme-ov-file#check-in

jbungenerflex commented:
HEllo @wiktor-flex , all your changes have been addressed. Let's go for a 3rd round of review. 
Thanks

jbungenerflex commented:
Hello @wiktor-flex, I have tested it on the "simple_airplane" test geometry. I changed a few things to make it metric ( initially I made this for a client that is in US units but I think it makes more sense to make it metric).

I fixed the log.info bug that you had spotted.
I tested both the "upload mesh and run" and the "download results" scripts
It now all works. LEt me know what you think. Thank you

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,126 @@
+
+    vm = project.volume_mesh  # get the volume mesh entity associated with that project.
+    log.info(f"The volume mesh contains the following boundaries:{vm.boundary_names}")
+    log.info(f"The mesh id is {vm.id}")
```
```suggestion
    log.info(f"The volume mesh ID is: {vm.id}")
```

wiktor-flex commented on examples/template_script_load_mesh_run_case_or_sweep.py:
```diff
@@ -0,0 +1,126 @@
+
+    # or if you want to simply launch the case\
+    project.run_case(params=params, name=f"case_name")
+    log.info(f"case id is {project.case.id}")
```
```suggestion
    log.info(f"The case ID is: {project.case.id}")
```

wiktor-flex reviewed (commented):
Mostly cosmetic changes for better readability, one change to how a project is setup to fix errors
